pages/products/[id].tsx

```diff
@@ -0,0 +1,224 @@
+import React, { useState, MouseEvent, ChangeEvent } from 'react';
+import Head from 'next/head';
+import Nav from '../../components/Nav';
+import Footer from '../../components/Footer';
+import Cookies from 'js-cookie';
+import Link from 'next/link';
+import { NextPageContext } from 'next';
+
+// using typescript: 	yarn add --dev typescript @types/node
+
+type Items = {
+  id: string;
+  name: string;
+  price: number;
+  src: string;
+  h2: string;
+  h3: string;
+  p: string;
+};
+type Props = { items: Items };
+
+// [id] => act as the part of the Path like 'localhost:3000/users/1(which is [id]'
+const Product = (props: Props) => {
+  const [price, setPrice] = useState<number>();
+  // typescript
+  // const [piece, setPiece] = useState<number | undefined>(undefined);
+  const [piece, setPiece] = useState(1);
+
+  // typescript error: Parameter 'e' implicitly has an 'any' type
+  function changePieces(e: ChangeEvent<HTMLInputElement>) {
+    setPiece(Number(e.target.value));
+  }
+
+  // after line 31, type of piece is updated into number
+  function showPrice() {
+    setPrice(piece * props.items.price);
+  }
+
+  if (!props.items) return <div>Item not found!</div>;
+
+  function goCart() {
+    const product = {
+      piece: piece,
+      price: price,
+      name: props.items.name,
+      src: props.items.src,
+      id: props.items.id,
+    };
+
+    const newCart = Cookies.getJSON('cart') || [];
+
+    newCart.push(product);
+
+    Cookies.set('cart', newCart);
+    window.location.reload();
+  }
+
+  return (
+    <div>
+      <Head>
+        <title>{props.items.name}</title>
+        <link rel="icon" href="/favicon.png" />
+      </Head>
+      <Nav />
+
+      <main>
+        <h1>{props.items.h2}</h1>
+        <h2>{props.items.name}</h2>
+        <img src={props.items.src} alt="items" />
```

Comment:
Same thing with the `alt` prop for the image - should be something that makes sense for this particular image. Right now, all product images will have the alt text of "items".